src/nfa/nfa.cc

```diff
@@ -355,40 +361,19 @@ void Nfa::add_trans(State state_from, Symbol symbol, State state_to) {
             state_transitions.insert(new_symbol_transitions);
         }
     }
-}
 
-void Nfa::add_trans(const State state_from, const Symbol symbol, const StateSet& states_to) {
-    // TODO: Define own exception.
-    if (!is_state(state_from)) {
-        throw std::out_of_range(std::to_string(state_from) + " is not a state.");
-    }
-    for (const State state_to: states_to) {
-        if (!is_state(state_to)) {
-            throw std::out_of_range(std::to_string(state_to) + " is not a state.");
-        }
-    }
+    if (state_from > max_state_)
+        max_state_ = state_from;
+    if (state_to > max_state_)
+        max_state_ = state_to;
+}
 
-    auto& state_transitions{delta[state_from] };
-    if (state_transitions.empty()) {
-        state_transitions.insert({ symbol, states_to });
-    } else if (state_transitions.back().symbol < symbol) {
-        state_transitions.insert({ symbol, states_to });
-    } else {
-        const auto symbol_transitions{ state_transitions.find(Move{symbol }) };
-        if (symbol_transitions != state_transitions.end()) {
-            // Add transitions with symbol already used on transitions from state_from.
-            symbol_transitions->insert(states_to);
-            return;
-        } else {
-            // Add transitions to a new Move struct with symbol yet unused on transitions from state_from.
-            const Move new_symbol_transitions {Move{symbol, states_to }};
-            state_transitions.insert(new_symbol_transitions);
-        }
+void Delta::remove(State src, Symbol symb, State tgt) {
+    if (src >= post.size()) {
+        throw std::runtime_error("State " + std::to_string(src) + " is not in Delta");
```

Comment:
throw exception or do nothing?

Reply:
I think it's matter of taste. I prefer throwing exception, it can help in debugging.